apps/arm/src/main.cpp

```diff
@@ -167,14 +168,75 @@ static mbed_error_status_t setToolTipDeployment(CANMsg &msg) {
   }
 }
 
-// Enable or disable PID tuning mode10
+// Enable or disable PID tuning mode
+// Manually send CAN message to SET_PID_TUNING_MODE with appropriate boolean
 static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
-  return MBED_SUCCESS;  // TODO
+  bool allowPIDTuning;
+  bool updateFailed = false;
+  msg.getPayload(allowPIDTuning);
+  if (allowPIDTuning) {
+    updateFailed |= turnTableActuator.allowPIDTuning();
+    updateFailed |= shoulderActuator.allowPIDTuning();
+    updateFailed |= elbowActuator.allowPIDTuning();
+    updateFailed |= wristLeftActuator.allowPIDTuning();
+    updateFailed |= wristRightActuator.allowPIDTuning();
+    updateFailed |= clawController.allowPIDTuning();
+  } else {
+    updateFailed |= turnTableActuator.disallowPIDTuning();
+    updateFailed |= shoulderActuator.disallowPIDTuning();
+    updateFailed |= elbowActuator.disallowPIDTuning();
+    updateFailed |= wristLeftActuator.disallowPIDTuning();
+    updateFailed |= wristRightActuator.disallowPIDTuning();
+    updateFailed |= clawController.disallowPIDTuning();
+  }
+  return updateFailed ? MBED_ERROR_MUTEX_LOCK_FAILED : MBED_SUCCESS;
 }
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-  return MBED_SUCCESS;  // TODO
+  printf("Received request to update PID params over CAN\n");
+  /** ------------------------------------
+   * | 32 bits  | 8 bits   |  8 bits     |
+   * | data     | vel/pos  |  Actuator ID|
+   *  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^*/
+  HWBRIDGE::ARM::PID::tuningApiPayload payload;
+  msg.getPayload(payload);
+  printf("payload: %s", HWBRIDGE::ARM::PID::str(payload).c_str());
+  LUT<HWBRIDGE::ARM::ACTUATORID, std::reference_wrapper<ActuatorController>> actuatorIDLUT = {
+      {HWBRIDGE::ARM::ACTUATORID::TURNTABLE, turnTableActuator},
+      {HWBRIDGE::ARM::ACTUATORID::SHOULDER, shoulderActuator},
+      {HWBRIDGE::ARM::ACTUATORID::ELBOW, elbowActuator},
+      {HWBRIDGE::ARM::ACTUATORID::WRISTLEFT, wristLeftActuator},
+      {HWBRIDGE::ARM::ACTUATORID::WRISTRIGHT, wristRightActuator},
+      {HWBRIDGE::ARM::ACTUATORID::CLAW, clawController}};
+  auto potentialTargetActuator = actuatorIDLUT[payload.actuatorID];
+  if (!potentialTargetActuator.has_value()) {
+    printf("ERROR: Invalid Actuator ID\n");
+    return MBED_ERROR_INVALID_ARGUMENT;
+  }
+  ActuatorController &targetActuator = potentialTargetActuator.value();
+  switch (msg.getID()) {
```

Comment:
@wmmc88 i tried using std::function, but that meant i had to use std::bind to bind the obj and the function but that was causing a weird error about an assertion failing. i didnt think to keep investigating bc i felt switch was more appropriate ofr this

Reply:
i dont remember the failing assertion, but i was binding the object and function to avoid code repition. I feel like this way is nicer and more legible. do u feel strongly that i should use a map for functions?